src/containers/Profile/Editprofile.js

```diff
@@ -0,0 +1,141 @@
+import React, { Component } from "react";
+import { connect } from "react-redux";
+
+import Modal from "../../components/Modal/Modal";
+import Input from "../../components/Input/Input";
+import Imageupload from "../../components/Imageupload/Imageupload";
+import { saveProfileInfo } from "../../redux/actions/profileActions/profileActions";
+import "./Styles/editprofile.css";
+
+export class UnconnectedEditprofile extends Component {
+  state = {
+    editprofile: false,
+    profiledata: this.props.profiledata
+  };
+
+  /**
+   * This method toggles the editprofile prop.
+   */
+  toggleEditProfile = event => {
+    this.setState(({ editprofile }) => ({
+      editprofile: !editprofile
+    }));
+  };
+
+  /**
+   * edit the profile information from the input
+   */
+  handleEditProfileData = event => {
+    const { name, value } = event.target;
+    this.setState(({ profiledata }) => ({
+      profiledata: {
+        ...profiledata,
+        [name]: value
+      }
+    }));
+  };
+
+  handleUpdateImage = image => {
+    this.setState(({ profiledata }) => ({
+      profiledata: {
+        ...profiledata,
+        image
+      }
+    }));
+  };
+
+  handleSubmitProfileData = event => {
+    const { saveProfileInfo } = this.props;
+    const { profiledata } = this.state;
+    event.preventDefault();
+    // close the modal
+    this.toggleEditProfile();
+    // submit the updated data
+    saveProfileInfo(profiledata);
+  };
+  render() {
+    const {
+      editprofile,
+      profiledata: { firstname = "", lastname = "", bio = "" }
+    } = this.state;
+    /**
+     *
+     * we have the attributes of profiledata
+     *  equal to an empty string incase the values are not provided
+     *
+     *
+     * */
+
+    return (
+      <div>
+        <a
+          data-test="activate-edit"
+          className="waves-effect orange darken-1 waves-light btn"
+          onClick={this.toggleEditProfile}
+        >
+          <i className="material-icons left">edit</i>Edit Profile
+        </a>
+        <Modal
```

Comment:
Decouple this modal here into a function. Render function shouldn't exceed 25lines.

Reply:
I've refactored this, now the render function is clean and small